### PR TITLE
[WIP] clientzoom: Ported client side zoom from server.

### PR DESCRIPTION
**Pull request by deekoder**

### CODE REVIEW

File: app/src/main/java/com/minio/io/alice/XDebug.java [hunk 1]
```diff
@@ -25,5 +25,5 @@
 
 public class XDebug {
     // set this to true during development only.
-    public static final boolean LOG = false;
+    public static final boolean LOG = true;
```
Comment (poornas) [inline app/src/main/java/com/minio/io/alice/XDebug.java]: toggle

**Comment by harshavardhana** [comment]
ping @deekoder sent a PR to your fork fixing conflicts - can you merge? https://github.com/deekoder/alice/pull/1

File: app/src/main/java/com/minio/io/alice/CustomFaceDetector.java [hunk 2]
```diff
@@ -114,4 +123,81 @@ public boolean setFocus(int id) {
...
-}
\ No newline at end of file
+    // Calculate optimal zoom factor depending on input factors
+    // such as Frame metadata and list of faces.
+    // This optimal zoom allows to control the zoom factor such that we
+    // we do not overflow our face bounding boxes.
+    private int calculateOptimalZoomFactor(Frame.Metadata metaData, SparseArray<Face> faces) {
+        Rect boundingBox = new Rect(0,0,metaData.getWidth(),metaData.getHeight());
+        Rect combinedRectangle = new Rect();
+        final int zoomOutBorderSize = 50;
```
Comment (fwessels) [inline app/src/main/java/com/minio/io/alice/CustomFaceDetector.java]: Maybe we should increase these border values, especially if we are running at high-res, 4K resolution
For 4K I would suggest quadrupling both zoomOutBorderSize and nozoomBorderSize

Comment (harshavardhana) [inline app/src/main/java/com/minio/io/alice/CustomFaceDetector.java]: :+1:

**Pull request merged**